src/fsharp/fsi/fsi.fs

```diff
@@ -613,13 +613,23 @@ type internal FsiCommandLineOptions(fsi: FsiEvaluationSessionHostConfig, argv: s
     let isInteractiveServer() = fsiServerName <> ""  
     let recordExplicitArg arg = explicitArgs <- explicitArgs @ [arg]
 
-    let executableFileName = 
-        lazy 
-            match tcConfigB.exename with
-            | Some s -> s
-            | None -> 
-            let currentProcess = System.Diagnostics.Process.GetCurrentProcess()
-            Path.GetFileName(currentProcess.MainModule.FileName)
+    let executableFileNameWithoutExtension =
+        lazy
+            let getFsiCommandLine () =
+                let fileNameWithoutExtension path = Path.GetFileNameWithoutExtension(path)
+
+                let currentProcess = System.Diagnostics.Process.GetCurrentProcess()
+                let processFileName = fileNameWithoutExtension currentProcess.MainModule.FileName
+
+                let commandLineExecutableFileName =
+                    try fileNameWithoutExtension (System.Environment.GetCommandLineArgs().[0])
```

Comment:
The reason for the try-catch block is that two things can go wrong:
1. `GetCommandLineArgs()` returns `null`, blowing up when trying to get the first item.
1. `GetCommandLineArgs()` returns an empty array, blowing up when trying to get the first item.